### PR TITLE
[ide-service] support docker auth IDE-164

### PR DESCRIPTION
**Pull request by iQQBot**
## Description

This PR adds support for private container registries for IDE images (for retrieving IDE config). Previously, we assumed all IDE images would be hosted publicly and would not require Docker authentication. 

## Related Issue(s)

Fixes [IDE-164](https://linear.app/gitpod/issue/IDE-164/[dogfood-cell]-ide-versions-do-not-resolve)

## How to test

It's too difficult to write a unit test, therefore we make sure that this change doesn't break existing implementation, and then we will test on our dogfood cell. cc @nandajavarma 

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - pd-ide-sereb5d2112a3</li>
	<li><b>🔗 URL</b> - <a href="https://pd-ide-sereb5d2112a3.preview.gitpod-dev.com/workspaces" target="_blank">pd-ide-sereb5d2112a3.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

**Comment by filiptronicek** [comment]
> Tested it not break preview env

can confirm, yay! 😆 

image

File: components/ide-service/pkg/server/server.go [hunk 1]
```diff
@@ -60,8 +68,38 @@ func Start(logger *logrus.Entry, cfg *config.ServiceConfiguration) error {
...
+			resolverOpts.Hosts = docker.ConfigureDefaultRegistries(
+				docker.WithAuthorizer(authorizerFromDockerConfig(dockerCfg)),
+			)
+		}
+
+		return docker.NewResolver(resolverOpts)
+	}
+	if cfg.DockerCfg != "" {
+		dockerCfg = loadDockerCfg(cfg.DockerCfg)
+		err = watch.File(ctx, cfg.DockerCfg, func() {
```
Comment (akosyakov) [inline components/ide-service/pkg/server/server.go:91]: @nandajavarma Is it good? I've seen some discussions today that we need to watch folder not a file? How do you do it for registry-facade?

**Comment by nandajavarma** [comment]
Thanks @akosyakov ! We are still debugging this issue currently ([internal](https://gitpod.slack.com/archives/C047WL9JBLG/p1686726059237529)). I would unblock this PR as soon as possible and iterate if that works.

**Comment by akosyakov** [comment]
@iQQBot I tried to start a workspace with VS Code Desktop but it opened VS Code browser.

**Comment by akosyakov** [comment]
I cannot smoke test workspaces, different things fails in VS Code Browser and Desktop:
Screenshot 2023-06-14 at 09 11 11

I doubt it is related to the PR though, something wrong with preview envs? cc @iQQBot fine to land anyway?

**Comment by laushinka** [comment]
> * workspaces are broken somehow, but i don't think it is related to this PR

@iQQBot Was this ok to land?

**Comment by akosyakov** [comment]
@laushinka I think so, see https://gitpod.slack.com/archives/C01KGM9BH54/p1686728600879949?thread_ts=1686664787.251019&cid=C01KGM9BH54

**Pull request merged**